tests/PHPStan/Analyser/data/infer-private-typed-property-type-from-constructor.php

```diff
@@ -0,0 +1,40 @@
+<?php // lint >= 7.4
+
+namespace InferPrivateTypedPropertyTypeFromConstructor;
+
+class Foo
+{
+	private GenericFoo $genericFoo;
+	private array $typedArray;
+
+	public function __construct(string ...$typedArray)
+	{
+		$this->genericFoo = $this->newGenericFoo(\stdClass::class);
+		$this->typedArray = $typedArray;
```

Comment:
I'm missing a test scenario here - the property has native type `int`, the constructor assigns type `string`. I'm interested in what type the final property has - should be `int`.

Reply:
Nice catch, I added some cases, it seems `TypehintHelper` is doing a very good job.

Do you see other scenario where it could misbehave?